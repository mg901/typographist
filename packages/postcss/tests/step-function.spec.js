const run = require('./run');
const { userConfig } = require('../../../mocks');

describe('transform step function', () => {
  it('should replace the step function with the font size in rem for each breakpoint', () => {
    const source = `
            h1 {
              font-size: step(6);
            }`;
    const compiled = `
            h1 {
              font-size: 2rem;
            }
@media (min-width: 48em) {
              h1 {
                            font-size: 2.6153846153846154rem;
              }
}
@media (min-width: 62em) {
              h1 {
                            font-size: 3.9310344827586206rem;
              }
}
@media (min-width: 75em) {
              h1 {
                            font-size: 3.9375rem;
              }
}`;
    
return run(source, compiled, userConfig);
  });
});
