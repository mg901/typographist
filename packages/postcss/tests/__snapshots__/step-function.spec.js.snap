// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step transform step 1`] = `
h1 {
  font-size: 2rem;
}
@media (min-width: 48em) {
  h1 {
    font-size: 2.6153846153846154rem;
  }
}
@media (min-width: 62em) {
  h1 {
    font-size: 3.9310344827586206rem;
  }
}
@media (min-width: 75em) {
  h1 {
    font-size: 3.9375rem;
  }
}
`;
